websockets/client.py

```diff
@@ -272,129 +273,153 @@ def handshake(self, wsuri, origin=None, available_extensions=None,
-    * ``compression`` is a shortcut to configure compression extensions;
-      by default it enables the "permessage-deflate" extension; set it to
-      ``None`` to disable compression
-
-    :func:`connect` raises :exc:`~websockets.uri.InvalidURI` if ``uri`` is
-    invalid and :exc:`~websockets.handshake.InvalidHandshake` if the opening
-    handshake fails.
+class Connect:
 
-    """
-    if loop is None:
-        loop = asyncio.get_event_loop()
-
-    # Backwards-compatibility: create_protocol used to be called klass.
-    # In the unlikely event that both are specified, klass is ignored.
-    if create_protocol is None:
-        create_protocol = klass
-
-    if create_protocol is None:
-        create_protocol = WebSocketClientProtocol
-
-    wsuri = parse_uri(uri)
-    if wsuri.secure:
-        kwds.setdefault('ssl', True)
-    elif kwds.get('ssl') is not None:
-        raise ValueError("connect() received a SSL context for a ws:// URI, "
-                         "use a wss:// URI to enable TLS")
-
-    if compression == 'deflate':
-        if extensions is None:
-            extensions = []
-        if not any(
-            extension_factory.name == ClientPerMessageDeflateFactory.name
-            for extension_factory in extensions
-        ):
-            extensions.append(ClientPerMessageDeflateFactory(
-                client_max_window_bits=True,
-            ))
-    elif compression is not None:
-        raise ValueError("Unsupported compression: {}".format(compression))
-
-    factory = lambda: create_protocol(
-        host=wsuri.host, port=wsuri.port, secure=wsuri.secure,
-        timeout=timeout, max_size=max_size, max_queue=max_queue,
-        read_limit=read_limit, write_limit=write_limit,
-        loop=loop, legacy_recv=legacy_recv,
-        origin=origin, extensions=extensions, subprotocols=subprotocols,
-        extra_headers=extra_headers,
-    )
-
-    if kwds.get('sock') is None:
-        host, port = wsuri.host, wsuri.port
-    else:
-        # If sock is given, host and port mustn't be specified.
-        host, port = None, None
-
-    transport, protocol = yield from loop.create_connection(
-        factory, host, port, **kwds)
-
-    try:
-        yield from protocol.handshake(
-            wsuri, origin=origin,
-            available_extensions=protocol.available_extensions,
-            available_subprotocols=protocol.available_subprotocols,
-            extra_headers=protocol.extra_headers,
+    def __init__(self, uri, *,
+                 create_protocol=None,
+                 timeout=10, max_size=2 ** 20, max_queue=2 ** 5,
+                 read_limit=2 ** 16, write_limit=2 ** 16,
+                 loop=None, legacy_recv=False, klass=None,
+                 origin=None, extensions=None, subprotocols=None,
+                 extra_headers=None, compression='deflate', **kwds):
+        """
+        This coroutine connects to a WebSocket server at a given ``uri``.
```

Comment:
That's slightly misleading because __init__ isn't — and cannot be — a coroutine (at least not until Python gets async object initialization, which I assume is a can of worms).

On Python ≥ 3.5, initializing a `Connect` instance returns an awaitable, so usage is effectively the same as a coroutine.

This problem actually dates back to the introduction to the `Connect` class. It predates this PR.